crates/bevy_render/src/render_resource/bind_group.rs

```diff
@@ -586,6 +590,15 @@ pub trait AsBindGroup {
         )
     }
 
+    /// Creates the bind group layout descriptor matching all bind groups returned by
+    /// [`AsBindGroup::as_bind_group_descriptor`]
+    fn bind_group_layout_descriptor() -> BindGroupLayoutDescriptor
+    where
+        Self: Sized,
+    {
+        todo!();
```

Comment:
TODO

Reply:
I think you can do similar to what the method right above this one does, something like
```suggestion
        BindGroupLayoutDescriptor::new(
            Self::label(),
            &Self::bind_group_layout_entries(render_device, false),
        )
```

Reply:
Ohh, the entries method takes render_device. Hmm. im not sure why that is

Reply:
@Zeophlite I think the way we should go about this is to remove the render device dependency in the other direction: we don't want to defer it but make it happen earlier, and inform as bind group by some other mechanism. For example, if we had a way of setting shader defs like BINDLESS and then checking them in as bind group, we can generate the right bind group layout etc for it. Because thats really what it is, shader-def-dependent bind group layouts.